pkg/config/tenancy/flags.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tenancy
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	tenancyEnabled = "multi_tenancy.enabled"
+	tenancyHeader  = "multi_tenancy.header"
+	validTenants   = "multi_tenancy.tenants"
+)
+
+// TenancyFlagsConfig describes which CLI flags for TLS client should be generated.
```

Comment:
the type does not seem necessary, we could just have global functions. Or, at least, it should be the same type as Options, that's the pattern we used, i.e. the config/options type exposes functions to initialize itself via CLI.